base_layer/wallet_ffi/src/lib.rs

```diff
@@ -6217,6 +6306,38 @@ mod test {
         }
     }
 
+    #[test]
+    pub fn test_mnemonic_word_lists() {
+        unsafe {
+            let mut error = 0;
+            let error_ptr = &mut error as *mut c_int;
+
+            for language in MnemonicLanguage::iterator() {
+                let language_str: *const c_char =
+                    CString::into_raw(CString::new(language.to_string()).unwrap()) as *const c_char;
+                let mnemonic_wordlist_ffi = mnemonic_word_list_for_language(language_str, error_ptr);
+                let mnemonic_wordlist = match *(language) {
+                    TariMnemonicLanguage::ChineseSimplified => mnemonic_wordlists::MNEMONIC_CHINESE_SIMPLIFIED_WORDS,
+                    TariMnemonicLanguage::English => mnemonic_wordlists::MNEMONIC_ENGLISH_WORDS,
+                    TariMnemonicLanguage::French => mnemonic_wordlists::MNEMONIC_FRENCH_WORDS,
+                    TariMnemonicLanguage::Italian => mnemonic_wordlists::MNEMONIC_ITALIAN_WORDS,
+                    TariMnemonicLanguage::Japanese => mnemonic_wordlists::MNEMONIC_JAPANESE_WORDS,
+                    TariMnemonicLanguage::Korean => mnemonic_wordlists::MNEMONIC_KOREAN_WORDS,
+                    TariMnemonicLanguage::Spanish => mnemonic_wordlists::MNEMONIC_SPANISH_WORDS,
+                };
+                assert_eq!(
```

Comment:
This test needs to be written as if it were being called from C.

Would look similar to:
```
let size = word_list_get_length(mnemonic_wordlist_ffi)

for (let i = 0; i < size; i++)
{
    let word = word_list_get_at(mnemonic_wordlist_ffi);
    asset_eq!(word, sample_wordlist[i]);
}
```